### ISSUE TITLE
Exception when retrieving all metadata

### ISSUE THREAD

**Issue by Bartheleway** [daishi4u/ForceCode#85]
Hi,

When trying to retrieve all metadata, I get this error :

```
[error] ZIP end of central directory record signature invalid (expects 0x06054b50, actually 0x8088b1f): Error: ZIP end of central directory record signature invalid (expects 0x06054b50, actually 0x8088b1f)
	at exports.Reader.Reader.readEndOfCentralDirectoryRecord (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\zip\zip.js:281:15)
	at exports.Reader.Reader.iterator (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\zip\zip.js:317:28)
	at exports.Reader.Reader.forEach (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\zip\zip.js:351:25)
	at \.vscode\extensions\brettwagner.forcecode-3.8.4\out\src\commands\retrieve.js:296:51
	at Parser.<anonymous> (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\xml2js\lib\parser.js:303:18)
	at emitOne (events.js:116:13)
	at Parser.emit (events.js:211:7)
	at SAXParser.onclosetag (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\xml2js\lib\parser.js:261:26)
	at emit (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\sax\lib\sax.js:624:35)
	at emitNode (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\sax\lib\sax.js:629:5)
	at closeTag (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\sax\lib\sax.js:889:7)
	at SAXParser.write (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\sax\lib\sax.js:1436:13)
	at Parser.exports.Parser.Parser.parseString (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\xml2js\lib\parser.js:322:31)
	at Parser.parseString (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\xml2js\lib\parser.js:5:59)
	at exports.parseString (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\xml2js\lib\parser.js:354:19)
	at \.vscode\extensions\brettwagner.forcecode-3.8.4\out\src\commands\retrieve.js:288:29
	at exports.Reader.Reader.forEach (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\zip\zip.js:363:15)
	at Readable.<anonymous> (\.vscode\extensions\brettwagner.forcecode-3.8.4\out\src\commands\retrieve.js:256:24)
	at emitNone (events.js:106:13)
	at Readable.emit (events.js:208:7)
	at endReadableNT (\.vscode\extensions\brettwagner.forcecode-3.8.4\node_modules\readable-stream\lib\_stream_readable.js:919:12)
	at _combinedTickCallback (internal/process/next_tick.js:138:11)
	at process._tickCallback (internal/process/next_tick.js:180:9)
```

I use :
- VS Code 1.27.1
- ForceCode 3.8.4

**Comment by daishi4u** [comment]
If it would be possible, could you try and retrieve all static resources only by doing it through the open function? Then let me know if you get this again. I believe your org may have a MIME type that isn't defined in my MIME dependency.

**Comment by Bartheleway** [comment]
So I get the exception when trying to open an "x-gzip" file.

**Comment by daishi4u** [comment]
Thanks that helps a lot 😀

**Comment by daishi4u** [comment]
This is now fixed in v3.8.5
